Bluetooth: Host: Fix missing term callback for PA syncs

Thalley opened the pull request:
When deleting a PA sync with bt_le_per_adv_sync_delete Zephyr should call the `term` callback for the PA sync as per the documentation for the callback.

This was not done in the case that the PA sync was terminated by local request.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Host: Fix missing term callback for PA syncs

When deleting a PA sync with bt_le_per_adv_sync_delete
Zephyr should call the `term` callback for the PA sync
as per the documentation for the callback.

This was not done in the case that the PA sync was
terminated by local request.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley commented:
CI is failing on something unrelated, please ignore the errors. Will re-run the checks once a fix has been merged

jori-nordic commented on subsys/bluetooth/host/scan.c:
```diff
@@ -1557,7 +1557,26 @@ int bt_le_per_adv_sync_delete(struct bt_le_per_adv_sync *per_adv_sync)
 		err = bt_le_per_adv_sync_terminate(per_adv_sync);
 
 		if (!err) {
+			const struct bt_le_per_adv_sync_term_info term_info = {
```
does it make sense to factor that out into a 'delete-with-cb' function (cf L917).

Thalley replied:
Did consider this myself; I'll add a function for it :)

Thalley commented on subsys/bluetooth/host/scan.c:
```diff
@@ -1557,7 +1557,26 @@ int bt_le_per_adv_sync_delete(struct bt_le_per_adv_sync *per_adv_sync)
+			SYS_SLIST_FOR_EACH_CONTAINER(&pa_sync_cbs, listener,
+						     node) {
+				if (listener->term) {
+					listener->term(per_adv_sync,
```
It should :) The pointer is equally valid, and allows the application to use it for cleanup or to even reuse it for another PA sync.

Thalley commented:
Updated the PR with a new function, `per_adv_sync_terminated`, that handles the calls to to the `term` callback and deletes the PA sync.

Thalley committed:
tests: Bluetooth: Update broadcast_iso test to support PA sync term cb

The callback is now called whenever the PA sync is terminated,
even if it was locally terminated. The test case has been
updated to handle that.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley commented:
Fixed failing test case that did not expect the PA term callback

Thalley committed:
Bluetooth: Host: Fix wrong addr for PA term when using adv list

If using the advertiser list when syncing to a PA, then the
address address needs to be updated for the PA sync object.
This is a small optimization to avoid doing a copy when
the address and SID is already set.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>
